fenix/app/src/main/java/org/mozilla/fenix/browser/TranslationsBinding.kt

```diff
@@ -36,37 +38,63 @@ class TranslationsBinding(
 ) : AbstractBinding<BrowserState>(browserStore) {
 
     override suspend fun onState(flow: Flow<BrowserState>) {
-        flow.mapNotNull { state -> state.findTab(sessionId) }.distinctUntilChangedBy {
-            it.translationsState
-        }.collect { sessionState ->
-            val translationsState = sessionState.translationsState
+        // Browser level flows
+        val browserFlow = flow.mapNotNull { state -> state }
+            .distinctUntilChangedBy {
+                it.translationEngine
+            }
 
-            if (translationsState.isTranslated) {
-                val fromSelected = translationsState.translationEngineState?.initialFromLanguage(
-                    translationsState.supportedLanguages?.fromLanguages,
-                )
-                val toSelected = translationsState.translationEngineState?.initialToLanguage(
-                    translationsState.supportedLanguages?.toLanguages,
+        // Session level flows
+        val sessionFlow = flow.mapNotNull { state -> state.findTab(sessionId) }
+            .distinctUntilChangedBy {
+                it.translationsState
+            }
+
+        // Applying the flows together
+        sessionFlow
+            .combine(browserFlow) { sessionState, browserState ->
+                TranslationsFlowState(
```

Comment:
@MatthewTighe Would also like your thoughts on this?

Reply:
Thanks for taking a look! I'll keep it the same for now.